solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -178,7 +178,7 @@ private Collector buildAndRunCollectorChain(QueryResult qr, Query query, Collect
     if (cmd.getSegmentTerminateEarly()) {
       final Sort cmdSort = cmd.getSort();
       final int cmdLen = cmd.getLen();
-      final Sort mergeSort = core.getSolrCoreState().getMergePolicySort();
+      final Sort mergeSort = core.getSolrCoreState().getMergePolicySort(); // TODO: need this account for indexSort also?
```

Comment:
https://issues.apache.org/jira/browse/SOLR-15390 is about deprecating the `segmentTerminateEarly` flag.